src/DataSource/Types.hs

```diff
@@ -457,11 +460,27 @@ class FromPageResultList a where
     fromPageResult :: Value -> Parser (PageResultList a)
 
 instance FromPageResultList TicketInfo where
-    fromPageResult = withObject "ticketList" $ \o ->
+    fromPageResult obj = asum
+        [ ticketListParser obj
+        , resultsTicketsParser obj
+        ]
+      where
```

Comment:
I was very interested in how you can implement this in a clean way. Thank you!

Reply:
:+1: